reana_job_controller/htcondorcern_job_manager.py

```diff
@@ -162,33 +162,27 @@ def _replace_absolute_paths_with_relative(self, cmd):
         return relative_paths_command
 
     def _format_arguments(self):
-        r"""Format HTCondor job execution arguments.
-
-        input  - ['bash', '-c',
-                  'cd /var/reana/users/00000000-0000-0000-0000-000000000000/
-                       workflows/e4691f78-25aa-4f90-9c3d-6873a97bdf16 ;
-                   python "code/helloworld.py" --inputfile "data/names.txt"
-                           --outputfile "results/greetings.txt" --sleeptime 0']
-        output - python \"code/helloworld.py\" --inputfile \"data/names.txt\"
-                 --outputfile \"results/greetings.txt\" --sleeptime 0
-        """
+        """Format HTCondor job execution arguments."""
         if self.workflow.type_ == "serial":
-            base_cmd = " ".join(self.cmd[2].split()[3:])
+            # Take only the user's command, removes the change directory to workflow workspace
+            # added by RWE-Serial since HTCondor implementation does not need it.
+            # E.g. "cd /path/to/workspace ; user-command" -> "user-command"
+            base_cmd = " ".join(self.cmd.split()[3:])
         elif self.workflow.type_ == "cwl":
             base_cmd = self.cmd[2].replace(self.workflow_workspace, "$_CONDOR_JOB_IWD")
         elif self.workflow.type_ == "yadage":
-            if "base64" in " ".join(self.cmd):
-                base_64_encoded_cmd = self.cmd[2].split("|")[0].split()[1]
+            if "base64" in self.cmd:
+                # E.g. echo ZWNobyAxCg==|base64 -d|bash
+                base_64_encoded_cmd = self.cmd.split("|")[0].split()[1]
```

Comment:
Same as the previous comment; simplified but not fully refactored.

Here the issue is related to the chosen interpreter to run the command. The [interpreter is embedded in the `cmd` string by workflow engine yadage](https://github.com/reanahub/reana-workflow-engine-yadage/blob/ae075c9e49555cc1efb59e682ef62be8e1f7ca77/reana_workflow_engine_yadage/externalbackend.py#L34) and this is strictly necessary, for example in [reanahub/reana-demo-JetMETAnalysis](https://github.com/alintulu/reana-demo-JetMETAnalysis/blob/0762860cdca7d41057ac56b3ea0729fcb9e557d6/workflow/steps.yaml#L29) where the interpreter must be `zsh`.

We should split the two entities, interpreter and command. Then add a new field in the `POST /jobs` endpoint, e.g. `interpreter`, specifying which interpreter to use. Each compute backend would implement then how to run the command in the specified interpreter.